opensearch-project/OpenSearch-Dashboards
# refactor: simplify theme configuration and defaulting

### Description

Reduces the amount of logic needed to add a new theme, reduces the number of files to be touched, + documents the process. There should be no change in experience.

Additionally, it appears we were persisting the label value in some places, while using the version string in others. This worked because we only ever checked that themeTag versions matched 'v7', but v8/next was the else case so it just so happened to work correctly. This tries to use the version string consistently, while keeping the label used for persisting to advanced settings (because I couldn't find an easy way to translate persisted values for existing configurations).

Unfortunately, due to need to run in different contexts, I had to write this to the least common denominator, so changes across the board are still necessary.

Note: There is an existing bug with browser based settings that doesn't respect the user selected setting in the topnav theme selector (even though the selected theme is actually defined). This isn't a released feature and not making things worse, so didn't address. Bug opened: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7689

### Issues Resolved

N/A

## Screenshot

| Context | Screenshot |
| ---- | ---- |
| v7 light | ![image](https://github.com/user-attachments/assets/c5495252-16af-4ce6-9773-d30da9deb29a)|
| v7 dark | ![image](https://github.com/user-attachments/assets/0043b24c-1605-4216-8ced-7097115bbb16)|
| v8 light | ![image](https://github.com/user-attachments/assets/0bd50604-bb34-44de-9874-8148569e53a2)| 
| v8 dark | ![image](https://github.com/user-attachments/assets/0249ad0e-e45e-4f3a-8b15-8ab2f5c632ee)| 

## Testing the changes

Most of the lines were already covered. The user header theme selection doesn't appear to be. Patch codecov was higher so I didn't add, but can if desired.

To test I:
- With user theme configuration turned off, made sure I could select the theme from advanced settings and see the correct result and css files were loaded as expected, and that returning to advanced settings the correct theme was selected
- With user theme configuration turned on, made sure sure I could select the theme from top nav and see correct result  and css files were loaded as expected, and that returning to the selector, the correct theme was selected
- Added a new theme (locally), and validated the above scenarios
- Bootstrapping/building locally still succeeds

## Changelog
- refactor: simplify theme configuration and defaulting
- deprecate: Deprecating `CssDistFilename` exports in favor of `themeCssDistFilenames` in `@osd/ui-shared-deps`

### Check List
